src/ufuncs.c

```diff
@@ -928,6 +929,211 @@ static void create_collection_func(char **args, npy_intp *dimensions,
+        else if (PyUnicode_Check(in1)) {
+            wkt = PyUnicode_AsUTF8(in1);
+            if (wkt == NULL) { return; }
+        } else {
+            PyErr_Format(PyExc_TypeError, "Expected bytes, found %s", in1->ob_type->tp_name);
+            return;
+        }
+
+        /* Read the WKT */
+        ret_ptr = GEOSWKTReader_read_r(context_handle, reader, wkt);
+        if (ret_ptr == NULL) {
+            return;
+        }
+
+        OUTPUT_Y;
+    }
+    GEOSWKTReader_destroy_r(context_handle, reader);
+}
+static PyUFuncGenericFunction from_wkt_funcs[1] = {&from_wkt_func};
+
+static char to_wkb_dtypes[2] = {NPY_OBJECT, NPY_OBJECT};
+static void to_wkb_func(char **args, npy_intp *dimensions,
+                        npy_intp *steps, void *data)
+{
+    void *context_handle = geos_context[0];
+    GEOSGeometry *in1;
+
+    GEOSWKBWriter *writer;
+    unsigned char *wkb;
+    size_t size;
+    
+    int dimension = 3;
+    int byte_order = 1;
+    char include_srid = 0;
+    char hex = 0;
+
+    /* Create the WKB writer */
+    writer = GEOSWKBWriter_create_r(context_handle);
+    if (writer == NULL) {
+        return;
+    }
+
+    GEOSWKBWriter_setOutputDimension_r(context_handle, writer, dimension);
+    GEOSWKBWriter_setByteOrder_r(context_handle, writer, byte_order);
+    GEOSWKBWriter_setIncludeSRID_r(context_handle, writer, include_srid);
+
+    UNARY_LOOP {
+        if (!get_geom(*(GeometryObject **)ip1, &in1)) { return; }
+        PyObject **out = (PyObject **)op1;
+
+        if (in1 == NULL) {  
+            Py_XDECREF(*out);
+            Py_INCREF(Py_None);
+            *out = Py_None;
+        } else {
+            if (hex) {
+                wkb = GEOSWKBWriter_writeHEX_r(context_handle, writer, in1, &size);
+            } else {
+                wkb = GEOSWKBWriter_write_r(context_handle, writer, in1, &size);
+            }
+            Py_XDECREF(*out);
+            *out = PyBytes_FromStringAndSize((char *) wkb, size);
+            GEOSFree_r(context_handle, wkb);
+        }
+    }
+    GEOSWKBWriter_destroy_r(context_handle, writer);
+}
+static PyUFuncGenericFunction to_wkb_funcs[1] = {&to_wkb_func};
+
+static char to_wkt_dtypes[2] = {NPY_OBJECT, NPY_OBJECT};
+static void to_wkt_func(char **args, npy_intp *dimensions,
+                        npy_intp *steps, void *data)
+{
+    void *context_handle = geos_context[0];
+    GEOSGeometry *in1;
+
+    GEOSWKTWriter *writer;
+    char *wkt;
+    char *format = "%s";
+    
+    char trim = 1;
+    int precision = 6;
+    int dimension = 3;
+    int use_old_3d = 0;
+
+    /* Create the WKT writer */
+    writer = GEOSWKTWriter_create_r(context_handle);
+    if (writer == NULL) {
+        return;
+    }
+
+    GEOSWKTWriter_setRoundingPrecision_r(context_handle, writer, precision);
+    GEOSWKTWriter_setTrim_r(context_handle, writer, trim);
+    GEOSWKTWriter_setOutputDimension_r(context_handle, writer, dimension);
+    GEOSWKTWriter_setOld3D_r(context_handle, writer, use_old_3d);
```

Comment:
I think these should be available as arguments to the functions. The only way I know to do that, is to actually add them to the ufunc inputs. So you'll get 5 input arguments. 

I do think they should e scalars, we don't want to call these 4 methods in the inner loop. I encountered the same problem in the buffer function, which I solved by asserting that it is a scalar.

Reply:
That means that you can't give them default values? (on the ufunc level, can still do that in the python wrapper)